extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -194,6 +198,43 @@ private RuntimeValue<GetterAccessorsContainer> implementPathParameterValueGetter
         return getterAccessorsContainer;
     }
 
+    /**
+     * Validates if the given classname contains a field `id` or annotated with `@Id`
+     *
+     * @throws IllegalStateException if the classname does not contain any sort of field identifier
+     */
+    private void validateClassHasFieldId(IndexView index, String entityType) {
+        // create a new independent class name that we can override
+        DotName className = DotName.createSimple(entityType);
+        List<FieldInfo> classFields = index.getClassByName(className).fields();
```

Comment:
You should cache the result of `index.getClassByName()` instead of calling it multiple times here and below.

Reply:
I think a better strategy would be to ask for fields named `id` and for annotations of the `jakarta.persistence.Id` type, rather than building a list of all fields and annotations. You can do this for each class in the hierarchy until you find one or you reach `Object`.